compiler/resolution/functionResolution.cpp

```diff
@@ -5864,6 +5864,16 @@ static void resolveNewHandleNonGenericInitializer(CallExpr*      call,
     if (isBlockStmt(call->parentExpr) == true) {
       call->insertBefore(def);
 
+      // Provide the return type for new expressions in default arguments to
+      // other parts of resolution. (Because the `new` expression is top level
+      // for argument symbols, we would fail to insert the temporary needed for
+      // argument analysis to appropriately determine the type without inserting
+      // this SymExpr directly)
+      if (isArgSymbol(call->parentSymbol) &&
+          toBlockStmt(call->parentExpr)->body.tail == call) {
+        call->insertAfter(new SymExpr(newTmp));
+      }
```

Comment:
Besides the question about at->isClass here, now with the comment, it's turning out to be enough lines it might be worth adding a function, where you can put the comment once.

Reply:
Classes don't run into this problem, because their "init" call is wrapped in a call to "_new", which does return the type.  But I definitely should make a function.

Reply:
So, the helper function I made currently doesn't have the isClass() check or a check if the parent of the call is a BlockStmt.  Both of these are necessary for the insertion to be valid, but the BlockStmt guard is already there for all the additions, and the isClass guard is there for all (albeit more obviously for 2 of the 3).  Should I move both of them into the helper and place the call at a more top level location in these functions?  Just the class one?

Reply:
Imma do both